addons/assigngear/functions/fn_assignGear.sqf

```diff
@@ -1,212 +1,167 @@
+ * Assigns a loadout defined in CfgLoadouts to a unit.
  */
 #include "\x\tmf\addons\assignGear\script_component.hpp"
-params [["_unit",objNull]];
-if (_unit isEqualTo objNull) exitWith {};
-// Exit if not local
-if !(local _unit) exitWith {};
-
-
-// TODO Remove backwards compatability (Date TBC)
 
-// Input type is array.
-if (_this isEqualType []) then {
-    if (count _this == 3) then {
-        params ["",
-            ["_faction",_unit getVariable [
-                QGVAR(faction),
-                toLower(faction _unit)
-            ]],
-            ["_role",_unit getVariable [QGVAR(role),"r"]]
-        ];
-        _unit setVariable [QGVAR(faction),_faction];
-        _unit setVariable [QGVAR(role),_role];
-
-    } else {
-        params ["",
-            ["_side",_unit getVariable [QGVAR(side),(side _unit) call EFUNC(common,sideToNum)]],
-            ["_faction",_unit getVariable [
-                QGVAR(faction),
-                toLower(faction _unit)
-            ]],
-            ["_role",_unit getVariable [QGVAR(role),"r"]]
-        ];
-        // Transform side ID into side
-        _unit setVariable [QGVAR(faction),_faction];
-        _unit setVariable [QGVAR(role),_role];
-        _unit setVariable [QGVAR(side),_side];
-    };
-};
+params [["_unit", player]];
 
-private _side = _unit getVariable [QGVAR(side),-1];
-private _faction = _unit getVariable [QGVAR(faction), toLower(faction _unit)];
-private _role = _unit getVariable [QGVAR(role),"r"];
-private _cfg = configNull;
-
-if (_side isEqualTo -1) then {
-    _cfg = missionConfigFile >> "cfgLoadouts" >> _faction;
-} else {
-    // OLD ASSIGN GEAR SYSTEM
-    _side = toLower([_side] call CFUNC(sideType));
-    _cfg = missionConfigFile >> "cfgLoadouts" >> _side >> _faction;
-};
-
-// Set variables to unit
-_unit setVariable [QGVAR(faction),_faction,true];
-_unit setVariable [QGVAR(role),_role,true];
-
-// Faction not found in mission try in modpack.
-if (!isClass _cfg) then {
-    _cfg = configFile >> "cfgLoadouts" >> _faction;
-};
+if !(local _unit) exitWith {};
 
-if (!isClass _cfg) exitWith {
-    // ERROR FACTION NOT FOUND.
-    private _message = format["ERROR: Faction %1 not found.",_faction];
-    if (is3DEN) then {
-        0 = [_message,1,5,true] spawn {
-            disableSerialization;
-            _this call BIS_fnc_3DENNotification;
+_this params [
+    "",
+    ["_faction", _unit getVariable [QGVAR(faction), toLower faction _unit]],
+    ["_role", _unit getVariable [QGVAR(role), "r"]]
+];
+
+// Sometimes in editor this function is run before preInit, this should make sure that the namespace exists
+private _namespace = missionNamespace getVariable [QGVAR(namespace), [FUNC(initNamespace)] call CBA_fnc_directCall];
+private _loadout = format ["loadout_%1_%2", _faction, _role];
+
+// Check if loadout if cached, if not then cache it
+private _loadoutArray = _namespace getVariable [_loadout, [_faction, _role] call FUNC(cacheAssignGear)];
```

Comment:
Note this will always call `[_faction, _role] call FUNC(cacheAssignGear)` as the array is evaluated before processing the `getVariable`